### PR TITLE
feat(ui): resource detail view

### PR DESCRIPTION
**Pull request by jeff-mccoy**
## Description

This PR introduces K8s detail views via a drawer component. The view is addressable as it also uses a route param, e.g.: `/workloads/pods/f49541dd-b571-49bd-9ce6-2824e015020a`. In addition to showing basic metadata, the raw yaml (un-styled currently) is also shown in a tabbed interface. Keyboard navigation is also provided:

- Up/Down: switching between resources in the table
- Left/Right: switching between tags: Metadata, YAML, Events
- Esc: close the details view

## Related Issue

- #34 

Screenshot 2024-08-01 at 3 27 47 AM

Screenshot 2024-08-01 at 3 31 48 AM

### CODE REVIEW

File: ui/src/lib/components/DataTable/component.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import type { KubernetesObject } from '@kubernetes/client-node'
+
+export async function loadResourceDetail(url: string, uid: string) {
```
Comment (TristanHoladay) [inline ui/src/lib/components/DataTable/component.ts:3]: broader question but this made me think of it. do we have some kind of error handling for network calls in place yet and if not do we want that atm?

Comment (jeff-mccoy) [reply ui/src/lib/components/DataTable/component.ts:3]: great question--we should add an issue for that, it should be addressed for sure

**Comment by decleaver** [comment]
created follow on issue: https://github.com/defenseunicorns/uds-runtime/issues/117

**Review by decleaver — APPROVED** [review]

**Comment by BillyFigueroa** [comment]
@jeff-mccoy do we want to keep the naming structure from flowbite to figma for this component? We have been using the term Drawer all along. Everything would be aligned if this was `components/Drawer/component.svelte`

Screenshot 2024-07-31 at 4 20 58 PM

Screenshot 2024-07-31 at 4 22 01 PM

**Comment by jeff-mccoy** [comment]
Sure we can rename it drawer

**Comment by UncleGedd** [comment]
Pulled it down, everything seems to work really well. I especially like the keyboard events

Were you able to use any Flowbite blocks to build this?

**Review by Madeline-UX — APPROVED** [review]
Passed UX/UI review

**Pull request merged**